From 20aae5137af839aa17b90ad9ea24531b08a2ab72 Mon Sep 17 00:00:00 2001
From: Martin Kosek <mkosek@redhat.com>
Date: Thu, 14 Mar 2013 14:36:39 +0100
Subject: [PATCH 82/98] Improve client install LDAP cert retrieval fallback

CA certificate retrieval function did not fallback from LDAP to
HTTP based retrieval in case of an LDAP error, when for example
GSSAPI authentication failed.

https://fedorahosted.org/freeipa/ticket/3512
---
 ipa-client/ipa-install/ipa-client-install | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ipa-client/ipa-install/ipa-client-install b/ipa-client/ipa-install/ipa-client-install
index e198d8dcb335d9d4ee91c70fb793b7026dba4004..61cf1395de80e04276b9dc5ae127f7ee5f33111b 100755
--- a/ipa-client/ipa-install/ipa-client-install
+++ b/ipa-client/ipa-install/ipa-client-install
@@ -1589,7 +1589,7 @@ def get_ca_cert(fstore, options, server, basedn):
                 except Exception, e:
                     os.unlink(ca_file)
                     raise
-            except errors.NoCertificateError, e:
+            except (errors.NoCertificateError, errors.LDAPError), e:
                 root_logger.debug(str(e))
                 url = http_url()
                 if existing_ca_cert:
-- 
1.8.3.1

